packages/@orbit/jsonapi/src/jsonapi-request-processor.ts

```diff
@@ -261,6 +261,8 @@ export default class JSONAPIRequestProcessor {
       if (this.responseHasContent(response)) {
         return response.json();
       }
+    } else if (response.status === 304) {
+      return Promise.resolve({ data: null });
```

Comment:
```suggestion
      return;
```

Reply:
I've applied this change and filled out the tests for both `query` and `pull` calls that result in a 304.